inc/msp/Client.hpp

```diff
@@ -126,111 +30,49 @@ enum MessageStatus {
 };
 
 struct ReceivedMessage {
-    uint8_t id;
-    std::vector<uint8_t> data;
+    uint32_t id;
+    ByteVector data;
     MessageStatus status;
 };
 
 class Client {
 public:
-    Client();
+    Client(const std::string &device, const size_t baudrate=115200);
 
     ~Client();
-
-    void setPrintWarnings(const bool warnings) {
-        print_warnings = warnings;
-    }
+    
+    void setDevice(const std::string& device);
+    std::string getDevice();
+    
+    void setBaudRate(const size_t& baud);
+    size_t getBaudRate();
+    
+    void printWarnings(const bool& val = true);
+    void printDebug(const bool& val = true);
+    
+    bool setVersion(int ver);
+    int getVersion();
+    
+    void setVariant(FirmwareVariant v);
+    FirmwareVariant getVariant();
 
     /**
      * @brief connect establish connection to serial device
      * @param device path or name of serial device
      * @param baudrate serial baudrate (default: 115200)
      * @return true on success
      */
-    void connect(const std::string &device, const size_t baudrate=115200);
-
-    /**
-     * @brief start starts the receiver thread that handles incomming messages
-     */
-    void start();
-
-    /**
-     * @brief stop stops the receiver thread
-     */
-    void stop();
-
-    /**
-     * @brief sendData send raw data and ID to flight controller, accepts any uint8 id
-     * @param id message ID
-     * @param data raw data
-     * @return true on success
-     * @return false on failure
-     */
-    bool sendData(const uint8_t id, const ByteVector &data = ByteVector(0));
-
-    /**
-     * @brief sendRequest request payload from FC
-     * @param id message ID
-     * @return true on success
-     * @return false on failure
-     */
-    bool sendRequest(const uint8_t id) {
-        return sendData(id, ByteVector());
-    }
-
-    bool sendRequest(const msp::ID id) {
-        return sendData(uint8_t(id), ByteVector());
-    }
-
-    /**
-     * @brief sendResponse send payload to FC
-     * @param response response with payload
-     * @return true on success
-     * @return false on failure
-     */
-    bool sendResponse(const msp::Response &response) {
-        return sendData(uint8_t(response.id()), response.encode());
-    }
+    bool connect();
+    bool disconnect();
 
-    /**
-     * @brief request requests payload from FC and block until payload has been received
-     * @param request request whose data will be set by the received payload
-     * @param timeout (optional) timeout in seconds
-     * @return true on success
-     * @return false on failure
-     * @return -1 on timeout
-     */
-    int request(msp::Request &request, const double timeout = 0);
+    
+    //synchronous message (wait for ack or response)
+    bool sendMessage(msp::Message& message, const double timeout = 0);
```

Comment:
Also update the README with the new API. E.g. replace the examples how to send formatted or raw messages.